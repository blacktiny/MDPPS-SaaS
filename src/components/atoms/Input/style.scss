.readonly-field {
  background-color: #f0f0f0 !important;
  cursor: default !important;
  pointer-events: none;
  font-size: 0.75rem !important;
  line-height: 1.42857143 !important;
  color: #14171a !important;
}
