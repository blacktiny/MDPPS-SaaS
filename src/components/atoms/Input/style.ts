import { Input as RInput } from 'rsuite'
import styled from '@emotion/styled'
import rfs, { boxModel } from 'utils/style'
import variables from 'assets/styles/variables'
import { css } from '@emotion/core'

const { Colors } = variables

export const DisabledStyle = css`
  background-color: #f0f0f0 !important;
  cursor: not-allowed;

  &::placeholder {
    color: ${Colors.Gray[200]};
    padding-left: 0 !important;
  }

  &:hover,
  &:focus {
    background-color: ${Colors.Gray[100]} !important;
    border: none;
  }
`

export const InvalidStyle = css`
  background-color: ${Colors.Gray[25]} !important;
  border: 1px solid ${Colors.Red[3]} !important;

  &:hover {
    background-color: ${Colors.Gray[25]} !important;
    border: 1px solid ${Colors.Red[3]} !important;
  }
`

export const InputStyle = css`
  background: ${Colors.Gray[50]} !important;
  border: 1px solid transparent !important;
  border-radius: 4px;
  font-size: ${rfs('12px')};
  min-height: ${boxModel('50px')};
  padding: ${boxModel('10px 20px')};
  transition: all 0.5s ease-in-out;
  width: 100%;
  color: ${Colors.Gray[500]};

  &::placeholder {
    color: ${Colors.Gray[400]};
    &:first-letter {
      text-transform: capitalize;
    }
  }

  &:hover {
    background: ${Colors.Gray[100]};
    border-color: transparent;
  }

  &:focus,
  &:active,
  &.active {
    border: 1px solid ${Colors.Blue[200]} !important;
    background: ${Colors.Gray[25]} !important;
  }
`

export const BaseInput = styled(RInput)<{ disabled?: boolean; invalid?: boolean }>`
  ${InputStyle}
  ${({ disabled }) => (disabled ? DisabledStyle : null)}
  ${({ invalid }) => (invalid ? InvalidStyle : null)}
`
