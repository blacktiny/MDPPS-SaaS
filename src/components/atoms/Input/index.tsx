import React, { useState, useEffect, ReactNode } from 'react'
import styled from '@emotion/styled'
import rfs, { boxModel, ErrorMsg } from '../../../utils/style'
import { Input as RInput } from 'rsuite'
import variables from '../../../assets/styles/variables'
import { InputProps } from 'rsuite/lib/Input/Input'
import { css } from '@emotion/core'
import loadable from '@loadable/component'
import FormFieldHeader from '../FormFieldHeader'
import EyeVisible from '../../../assets/icons/nav-eye.svg'
import EyeInvisible from '../../../assets/icons/eye-disabled.svg'
import { FieldElement } from 'react-hook-form'
import className from 'classnames'
import './style.scss'

const PasswordStrengthBar = loadable(() => import('react-password-strength-bar'))

const { Colors, Fonts } = variables

const DisabledStyle = css`
  .control-label {
    color: ${Colors.Gray[200]};
  }
  .base-input {
    background-color: #f0f0f0;
    cursor: not-allowed;
    &::placeholder {
      color: ${Colors.Gray[200]};
      padding-left: 0 !important;
    }
    &:hover,
    &:focus {
      background-color: ${Colors.Gray[100]} !important;
      border: none;
    }
  }
  .help-block {
    color: ${Colors.Gray[200]};
  }
`
const InvalidStyle = css`
  .base-input {
    background-color: ${Colors.Gray[25]};
    border: 1px solid ${Colors.Red[3]};
    color: ${Colors.Red[3]};
    &:hover {
      background-color: ${Colors.Gray[25]};
      border: 1px solid ${Colors.Red[3]};
    }
  }
`

const BaseInput = styled(RInput)`
  background: ${Colors.Gray[50]};
  border: 1px solid transparent;
  border-radius: 4px;
  font-size: ${rfs('12px')};
  height: ${boxModel('50px')};
  padding: ${boxModel('10px 20px')};
  transition: all 0.5s ease-in-out;
  width: 100%;
  color: ${Colors.Gray[500]};

  &::placeholder {
    color: ${Colors.Gray[400]};
    &:first-letter {
      text-transform: capitalize;
    }
  }

  &:hover {
    background: ${Colors.Gray[100]};
    border-color: transparent;
  }
  &:focus,
  &:active {
    &:hover {
      background: ${Colors.Gray[25]} !important;
    }
    border: 1px solid ${Colors.Blue[200]};
    background: ${Colors.Gray[25]} !important;
  }
  &[disabled] {
    background: #f0f0f0 !important;
  }
`
const InputContainer = styled('div')<{ disabled?: boolean; invalid?: boolean }>`
  ${({ disabled }) => (disabled ? DisabledStyle : null)}
  ${({ invalid }) => (invalid ? InvalidStyle : null)}
  .password-strength-bar {
    border-radius: 2px;
    div[style='width: 4px;'] {
      display: none;
    }
    & > div {
      margin-top: 12px !important;
      max-width: 13.625rem;
      & > div {
        transition: 0.5s ease;
        height: 10px !important;
      }
    }
    p {
      margin: 0 !important;
      position: absolute;
      right: 0;
      top: -5px;
      font-size: ${rfs(Fonts.Size.XXSmall)} !important;
    }
    &.very-weak > div > div:first-of-type {
      background-color: ${Colors.Red[3]} !important;
    }
  }
`

const InputGroup = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  border-radius: 3px;
`
const PasswordVisibilityToggle = styled.button`
  position: absolute;
  right: ${rfs('20px')};
  z-index: 2;
  background: transparent;
  padding: 0;
  font-size: 0;
  &:focus {
    outline: none;
  }
  img {
    width: 1.25rem;
  }
`

interface Props extends InputProps {
  placeholder?: string
  name: string
  invalid?: boolean
  disabled?: boolean
  type?: string
  errormsg?: string
  text?: string | ReactNode
  ref?: React.Ref<FieldElement | Element>
  updateScoreRef?: (value: number) => void
  defaultValue?: string
  onChange?: (value?) => void
  onFocus?: () => void
  onBlur?: () => void
  showErrorWrapper?: boolean
  characters?: string
  testErrorId?: string
  disableLabel?: boolean
  setReadOnlyStyle?: boolean
  containerClassName?: string
  inputClassName?: string
  required?: boolean
  value?: string
  maxLength?: number
}

const Input: React.FC<Props> = React.forwardRef(
  (
    {
      name,
      label,
      placeholder,
      disabled,
      invalid,
      type,
      showPasswordStatus,
      errormsg,
      text,
      updateScoreRef,
      defaultValue,
      onChange,
      onFocus,
      onBlur,
      showErrorWrapper,
      characters,
      testErrorId,
      disableLabel = true,
      setReadOnlyStyle,
      containerClassName,
      inputClassName,
      required,
      value,
      maxLength
    },
    ref
  ) => {
    const [password, setPassword] = useState('')
    const [passwordStrengthScore, setPasswordStrengthScore] = useState(0)
    const [passwordVisibility, setPasswordVisibility] = useState(false)

    const updatePasswordStrengthScore = (score: number) => {
      updateScoreRef(score)
      setPasswordStrengthScore(score)
    }

    const onChangeInput = value => {
      if (onChange) {
        onChange(value)
      }
    }

    const onBlurInput = () => {
      if (onBlur) {
        onBlur()
      }
    }

    const onFocusInput = () => {
      if (onFocus) {
        onFocus()
      }
    }

    useEffect(() => {
      if (characters === '' || characters) setPassword(characters)
    }, [characters])

    const baseInputStyle = className({
      'base-input': true,
      'readonly-field': setReadOnlyStyle
    })
      .concat(' ')
      .concat(inputClassName)

    return (
      <InputContainer invalid={invalid} disabled={disableLabel && disabled} className={containerClassName}>
        <FormFieldHeader label={label} name={name} text={text} required={required} />
        <InputGroup>
          <BaseInput
            id={name}
            name={name}
            value={value}
            placeholder={placeholder}
            className={baseInputStyle}
            disabled={disabled}
            type={type === 'password' ? (passwordVisibility ? 'text' : type) : type}
            onChange={onChangeInput}
            onBlur={onBlurInput}
            onFocus={onFocusInput}
            inputRef={ref}
            autoComplete={type === 'password' ? 'new-password' : 'off'}
            defaultValue={defaultValue}
            maxLength={maxLength}
          />
          {type === 'password' && (
            <PasswordVisibilityToggle
              className="visibility"
              onClick={() => {
                setPasswordVisibility(!passwordVisibility)
              }}
              type="button"
            >
              <img src={passwordVisibility ? EyeInvisible : EyeVisible} alt="toggle-visibility" />
            </PasswordVisibilityToggle>
          )}
        </InputGroup>
        <div className={showErrorWrapper ? 'error-wrapper' : ''}>
          {errormsg && (
            <ErrorMsg data-testid={testErrorId} className={!showErrorWrapper ? '' : 'my-0'}>
              {errormsg}
            </ErrorMsg>
          )}
        </div>
        {type === 'password' && password.length > 0 && showPasswordStatus && (
          <PasswordStrengthBar
            password={password}
            className={`password-strength-bar${passwordStrengthScore === 0 ? ' very-weak' : ''}`}
            shortScoreWord="Weak"
            scoreWords={['Weak', 'Weak', 'Moderate', 'Good', 'Strong']}
            barColors={[Colors.Gray[100], Colors.Orange[200], Colors.Orange[150], Colors.Blue[150], Colors.Green[300]]}
            minLength={8}
            onChangeScore={updatePasswordStrengthScore}
          />
        )}
      </InputContainer>
    )
  }
)

Input.displayName = 'Input Container'
export default Input
