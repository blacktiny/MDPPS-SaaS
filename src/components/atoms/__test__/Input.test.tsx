import React from 'react'
import { render } from '@testing-library/react'
import Input from '../Input'

test('should render children correctly', () => {
  const { getByLabelText } = render(
    <Input label="User Name" name="username" placeholder="Enter username" />
  )

  expect(getByLabelText('User Name')).toBeInTheDocument()
})

test('should look as expected in the NORMAL state', () => {
  const { baseElement } = render(
    <Input label="User Name" name="username" placeholder="Enter username" />
  )
  expect(baseElement).toMatchSnapshot()
})

test('should look as expected in the Invalid state', () => {
  const { baseElement } = render(
    <Input
      label="User Name"
      name="username"
      placeholder="Enter username"
      invalid
    />
  )
  expect(baseElement).toMatchSnapshot()
})

test('should look as expected in the DISABLED state', () => {
  const { baseElement } = render(
    <Input
      label="User Name"
      name="username"
      placeholder="Enter username"
      disabled
    />
  )

  expect(baseElement).toMatchSnapshot()
})
