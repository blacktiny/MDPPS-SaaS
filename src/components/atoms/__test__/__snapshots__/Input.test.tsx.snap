// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should look as expected in the DISABLED state 1`] = `
.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0.375rem;
}

.emotion-0 {
  color: #000000;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.02438rem;
  -moz-letter-spacing: 0.02438rem;
  -ms-letter-spacing: 0.02438rem;
  letter-spacing: 0.02438rem;
  line-height: 1.6;
  margin-bottom: 0;
}

.emotion-0::first-letter {
  text-transform: capitalize;
}

.emotion-1 {
  font-size: 0.75rem;
  -webkit-letter-spacing: 0.01375rem;
  -moz-letter-spacing: 0.01375rem;
  -ms-letter-spacing: 0.01375rem;
  letter-spacing: 0.01375rem;
  line-height: 1.5;
  color: #98a0ac;
  min-height: auto;
}

.emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
}

.emotion-3 {
  background: #F9F9FA;
  border: 1px solid transparent;
  border-radius: 4px;
  font-size: 0.75rem;
  height: 3.125rem;
  padding: 0.625rem 1.25rem;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  width: 100%;
  color: #000000;
}

.emotion-3::placeholder {
  color: #98a0ac;
}

.emotion-3::-webkit-input-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3::-moz-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3:-ms-input-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3::placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3:hover {
  background: #F2F2F2;
  border-color: transparent;
}

.emotion-3:focus,
.emotion-3:active {
  border: 1px solid #4284FC;
  background: #FFFFFF !important;
}

.emotion-3:focus:hover,
.emotion-3:active:hover {
  background: #FFFFFF !important;
}

.emotion-3[disabled] {
  background: #f0f0f0 !important;
}

.emotion-5 .control-label {
  color: #AAB8C2;
}

.emotion-5 .base-input {
  background-color: #f0f0f0;
  cursor: not-allowed;
}

.emotion-5 .base-input::-webkit-input-placeholder {
  color: #AAB8C2;
  padding-left: 0 !important;
}

.emotion-5 .base-input::-moz-placeholder {
  color: #AAB8C2;
  padding-left: 0 !important;
}

.emotion-5 .base-input:-ms-input-placeholder {
  color: #AAB8C2;
  padding-left: 0 !important;
}

.emotion-5 .base-input::placeholder {
  color: #AAB8C2;
  padding-left: 0 !important;
}

.emotion-5 .base-input:hover,
.emotion-5 .base-input:focus {
  background-color: #F2F2F2 !important;
  border: none;
}

.emotion-5 .help-block {
  color: #AAB8C2;
}

.emotion-5 .password-strength-bar {
  border-radius: 2px;
}

.emotion-5 .password-strength-bar div[style='width: 4px;'] {
  display: none;
}

.emotion-5 .password-strength-bar > div {
  margin-top: 12px !important;
  max-width: 13.625rem;
}

.emotion-5 .password-strength-bar > div > div {
  -webkit-transition: 0.5s ease;
  transition: 0.5s ease;
  height: 10px !important;
}

.emotion-5 .password-strength-bar p {
  margin: 0 !important;
  position: absolute;
  right: 0;
  top: -5px;
  font-size: 0.75rem !important;
}

.emotion-5 .password-strength-bar.very-weak > div > div:first-of-type {
  background-color: #d9001b !important;
}

<body>
  <div>
    <div
      class="emotion-5"
      disabled=""
    >
      <div
        class="input-header emotion-2"
      >
        <label
          class="rs-control-label control-label emotion-0"
          for="username"
        >
          <span>
            User Name
            <span
              style="color: red;"
            />
          </span>
        </label>
        <span
          class="rs-help-block help-block emotion-1"
        />
      </div>
      <div
        class="emotion-4"
      >
        <input
          autocomplete="off"
          class="rs-input base-input undefined emotion-3"
          disabled=""
          id="username"
          name="username"
          placeholder="Enter username"
          type="text"
          value=""
        />
      </div>
      <div
        class=""
      />
    </div>
  </div>
</body>
`;

exports[`should look as expected in the Invalid state 1`] = `
.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0.375rem;
}

.emotion-0 {
  color: #000000;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.02438rem;
  -moz-letter-spacing: 0.02438rem;
  -ms-letter-spacing: 0.02438rem;
  letter-spacing: 0.02438rem;
  line-height: 1.6;
  margin-bottom: 0;
}

.emotion-0::first-letter {
  text-transform: capitalize;
}

.emotion-1 {
  font-size: 0.75rem;
  -webkit-letter-spacing: 0.01375rem;
  -moz-letter-spacing: 0.01375rem;
  -ms-letter-spacing: 0.01375rem;
  letter-spacing: 0.01375rem;
  line-height: 1.5;
  color: #98a0ac;
  min-height: auto;
}

.emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
}

.emotion-3 {
  background: #F9F9FA;
  border: 1px solid transparent;
  border-radius: 4px;
  font-size: 0.75rem;
  height: 3.125rem;
  padding: 0.625rem 1.25rem;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  width: 100%;
  color: #000000;
}

.emotion-3::placeholder {
  color: #98a0ac;
}

.emotion-3::-webkit-input-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3::-moz-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3:-ms-input-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3::placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3:hover {
  background: #F2F2F2;
  border-color: transparent;
}

.emotion-3:focus,
.emotion-3:active {
  border: 1px solid #4284FC;
  background: #FFFFFF !important;
}

.emotion-3:focus:hover,
.emotion-3:active:hover {
  background: #FFFFFF !important;
}

.emotion-3[disabled] {
  background: #f0f0f0 !important;
}

.emotion-5 .base-input {
  background-color: #FFFFFF;
  border: 1px solid #d9001b;
  color: #d9001b;
}

.emotion-5 .base-input:hover {
  background-color: #FFFFFF;
  border: 1px solid #d9001b;
}

.emotion-5 .password-strength-bar {
  border-radius: 2px;
}

.emotion-5 .password-strength-bar div[style='width: 4px;'] {
  display: none;
}

.emotion-5 .password-strength-bar > div {
  margin-top: 12px !important;
  max-width: 13.625rem;
}

.emotion-5 .password-strength-bar > div > div {
  -webkit-transition: 0.5s ease;
  transition: 0.5s ease;
  height: 10px !important;
}

.emotion-5 .password-strength-bar p {
  margin: 0 !important;
  position: absolute;
  right: 0;
  top: -5px;
  font-size: 0.75rem !important;
}

.emotion-5 .password-strength-bar.very-weak > div > div:first-of-type {
  background-color: #d9001b !important;
}

<body>
  <div>
    <div
      class="emotion-5"
    >
      <div
        class="input-header emotion-2"
      >
        <label
          class="rs-control-label control-label emotion-0"
          for="username"
        >
          <span>
            User Name
            <span
              style="color: red;"
            />
          </span>
        </label>
        <span
          class="rs-help-block help-block emotion-1"
        />
      </div>
      <div
        class="emotion-4"
      >
        <input
          autocomplete="off"
          class="rs-input base-input undefined emotion-3"
          id="username"
          name="username"
          placeholder="Enter username"
          type="text"
          value=""
        />
      </div>
      <div
        class=""
      />
    </div>
  </div>
</body>
`;

exports[`should look as expected in the NORMAL state 1`] = `
.emotion-5 .password-strength-bar {
  border-radius: 2px;
}

.emotion-5 .password-strength-bar div[style='width: 4px;'] {
  display: none;
}

.emotion-5 .password-strength-bar > div {
  margin-top: 12px !important;
  max-width: 13.625rem;
}

.emotion-5 .password-strength-bar > div > div {
  -webkit-transition: 0.5s ease;
  transition: 0.5s ease;
  height: 10px !important;
}

.emotion-5 .password-strength-bar p {
  margin: 0 !important;
  position: absolute;
  right: 0;
  top: -5px;
  font-size: 0.75rem !important;
}

.emotion-5 .password-strength-bar.very-weak > div > div:first-of-type {
  background-color: #d9001b !important;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0.375rem;
}

.emotion-0 {
  color: #000000;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.02438rem;
  -moz-letter-spacing: 0.02438rem;
  -ms-letter-spacing: 0.02438rem;
  letter-spacing: 0.02438rem;
  line-height: 1.6;
  margin-bottom: 0;
}

.emotion-0::first-letter {
  text-transform: capitalize;
}

.emotion-1 {
  font-size: 0.75rem;
  -webkit-letter-spacing: 0.01375rem;
  -moz-letter-spacing: 0.01375rem;
  -ms-letter-spacing: 0.01375rem;
  letter-spacing: 0.01375rem;
  line-height: 1.5;
  color: #98a0ac;
  min-height: auto;
}

.emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
}

.emotion-3 {
  background: #F9F9FA;
  border: 1px solid transparent;
  border-radius: 4px;
  font-size: 0.75rem;
  height: 3.125rem;
  padding: 0.625rem 1.25rem;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  width: 100%;
  color: #000000;
}

.emotion-3::placeholder {
  color: #98a0ac;
}

.emotion-3::-webkit-input-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3::-moz-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3:-ms-input-placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3::placeholder:first-letter {
  text-transform: capitalize;
}

.emotion-3:hover {
  background: #F2F2F2;
  border-color: transparent;
}

.emotion-3:focus,
.emotion-3:active {
  border: 1px solid #4284FC;
  background: #FFFFFF !important;
}

.emotion-3:focus:hover,
.emotion-3:active:hover {
  background: #FFFFFF !important;
}

.emotion-3[disabled] {
  background: #f0f0f0 !important;
}

<body>
  <div>
    <div
      class="emotion-5"
    >
      <div
        class="input-header emotion-2"
      >
        <label
          class="rs-control-label control-label emotion-0"
          for="username"
        >
          <span>
            User Name
            <span
              style="color: red;"
            />
          </span>
        </label>
        <span
          class="rs-help-block help-block emotion-1"
        />
      </div>
      <div
        class="emotion-4"
      >
        <input
          autocomplete="off"
          class="rs-input base-input undefined emotion-3"
          id="username"
          name="username"
          placeholder="Enter username"
          type="text"
          value=""
        />
      </div>
      <div
        class=""
      />
    </div>
  </div>
</body>
`;
